modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/FilePageStore.java

```diff
@@ -59,7 +61,16 @@
     public static final int HEADER_SIZE = 8/*SIGNATURE*/ + 4/*VERSION*/ + 1/*type*/ + 4/*page size*/;
 
     /** */
-    private final File cfgFile;
+    private volatile File cfgFile;
+
+    /** */
+    private final IgniteOutClosure<File> fileProvider;
+
+    /**
+     * Caches the existance state of storage file. After it is initialized, it will be not set to null
+     * during FilePageStore lifecycle.
+     */
+    private volatile Boolean fileExists;
```

Comment:
The same as above. We could calculate it all the time when it would be necessary.

Reply:
We still need the field caching the flag of file existance to reduce file system calls.